cli/src/commands/developer/execute.rs

```diff
@@ -33,56 +37,69 @@ use snarkvm::{
 };
 
 use aleo_std::StorageMode;
-use anyhow::{Result, anyhow, bail};
-use clap::Parser;
+use anyhow::{Context, Result, bail};
+use clap::{Parser, builder::NonEmptyStringValueParser};
 use colored::Colorize;
 use std::{path::PathBuf, str::FromStr};
+use tracing::debug;
+use ureq::http::Uri;
 use zeroize::Zeroize;
 
 /// Executes an Aleo program function.
 #[derive(Debug, Parser)]
+#[command(
+    group(clap::ArgGroup::new("mode").required(true).multiple(false)),
+    group(clap::ArgGroup::new("key").required(true).multiple(false))
+)]
 pub struct Execute {
     /// The program identifier.
+    #[clap(value_parser=NonEmptyStringValueParser::default())]
     program_id: String,
     /// The function name.
+    #[clap(value_parser=NonEmptyStringValueParser::default())]
     function: String,
     /// The function inputs.
     inputs: Vec<String>,
-    /// Specify the network to create an execution for.
-    #[clap(default_value = "0", long = "network")]
-    pub network: u16,
     /// The private key used to generate the execution.
-    #[clap(short, long)]
+    #[clap(short = 'p', long, group = "key", value_parser=NonEmptyStringValueParser::default())]
     private_key: Option<String>,
     /// Specify the path to a file containing the account private key of the node
-    #[clap(long)]
+    #[clap(long, group = "key", value_parser=NonEmptyStringValueParser::default())]
     private_key_file: Option<String>,
-    /// The endpoint to query node state from.
-    #[clap(short, long)]
-    query: String,
+    /// Use a developer validator key to generate the execution
+    #[clap(long, group = "key")]
+    dev_key: Option<u16>,
+    /// The endpoint to query node state from and broadcast to (if set to broadcast).
+    #[clap(short, long, default_value=DEFAULT_ENDPOINT)]
+    endpoint: Uri,
     /// The priority fee in microcredits.
-    #[clap(long)]
-    priority_fee: Option<u64>,
+    #[clap(long, default_value_t = 0)]
+    priority_fee: u64,
     /// The record to spend the fee from.
     #[clap(short, long)]
     record: Option<String>,
     /// The endpoint used to broadcast the generated transaction.
-    #[clap(short, long, conflicts_with = "dry_run")]
-    broadcast: Option<String>,
+    #[clap(short, long, group = "mode")]
+    broadcast: bool,
```

Comment:
Moreover, having a separate query and broadcast endpoint is a privacy feature. Because you can query private state (records) from one node, and broadcast the transaction to another.